packages/dom/src/dom/is-text-field.js

```diff
@@ -1,15 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import isHTMLInputElement from './is-html-input-element';
+
 /**
  * Check whether the given element is a text field, where text field is defined
  * by the ability to select within the input, or that it is contenteditable.
  *
  * See: https://html.spec.whatwg.org/#textFieldSelection
  *
- * @param {HTMLElement} element The HTML element.
+ * @param {Element} element The HTML element.
  *
  * @return {boolean} True if the element is an text field, false if not.
```

Comment:
This boolean result narrows us to at least `HTMLElement`, right?
```suggestion
 * @return {element is HTMLElement} True if the element is an text field, false if not.
```